### PR TITLE
[Documentation] Concepts: Shipments

### PR DESCRIPTION
**Pull request by TheMadeleine**
| Q               | A
| --------------- | ---
| Doc fix?        | yes
| New docs?    | no
| BC breaks?      | no
| Related tickets | partially #5275 
| License         | MIT

### CODE REVIEW

**Comment by TheMadeleine** [comment]
Waiting for a helping hand with the calculators here. :)

File: docs/book/shipments.rst [hunk 1]
```diff
@@ -4,17 +4,75 @@
...
+
+The allowed transitions between these states are:
+
+.. code-block:: yaml
+
+   transitions:
+      request_shipping:
+          from: [cart]
+          to: ready
+      cancel:
```
Comment (lchrusciel) [inline docs/book/shipments.rst]: It is something for later, but I guess we are missing a transition between `partially_shipped` and `cancelled`

File: docs/book/shipments.rst [hunk 2]
```diff
@@ -4,17 +4,75 @@
...
 Shipping Methods
 ----------------
 
-...
+**ShippingMethod** in Sylius is an entity that represent the way an order can be shipped to a customer.
+
+How to create a ShippingMethod programmatically?
+''''''''''''''''''''''''''''''''''''''''''''''''
+
+As usually use a factory to create a new ShippingMethod. Give it a ``code``, set a desired shipping calculator, and set a ``zone``.
```
Comment (NeverResponse) [inline docs/book/shipments.rst]: It would be nice to know the significance of **zone**.

File: docs/book/shipments.rst [hunk 3]
```diff
@@ -4,49 +4,139 @@
 Shipments
 =========
 
-...
+A **Shipment** is a representation of a shipping request for an Order. Sylius supports multiple shipments per one Order.
+
+How to create a Shipment programmatically?
```
Comment (pjedrzejewski) [inline docs/book/shipments.rst]: I'd remove this part, as it is not that simple, it works, but the structure is not correct. We should use the processor to create shipments for an order, but we can document it later.

File: docs/book/shipments.rst [hunk 4]
```diff
@@ -4,49 +4,139 @@
...
+
+As usually use a factory to create a new ShippingMethod. Give it a ``code``, set a desired shipping calculator, and set a ``zone``.
+At the end add it to the system using a repository.
+
+.. code-block:: php
+
+    $shippingMethod = $this->container->get('sylius.factory.shipping_method')->createNew();
+
+    $shippingMethod->setCode('DHL');
+    $shippingMethod->setCalculator(DefaultCalculators::FLAT_RATE);
```
Comment (pjedrzejewski) [inline docs/book/shipments.rst]: We are missing configuration:

``` php
$shippingMethod->setConfiguration(['amount' => 500]);
```

File: docs/book/shipments.rst [hunk 5]
```diff
@@ -4,49 +4,139 @@
...
-...
+To select a proper service we have a one that decides for us
+- the `DelegatingCalculator <https://github.com/Sylius/Sylius/blob/master/src/Sylius/Component/Shipping/Calculator/DelegatingCalculator.php>`_.
+Basing on the **ShippingMethod** assigned on the Shipment it will get its calculator type and configuration and calculate the cost properly.
 
-Final Thoughts
---------------
+.. code-block:: php
+
+    $this->container->get('sylius.repository.shipment')->add($shipment);
```
Comment (pjedrzejewski) [inline docs/book/shipments.rst]: This should be removed.

File: docs/book/shipments.rst [hunk 6]
```diff
@@ -4,49 +4,139 @@
...
+* **FlexibleRateCalculator** - on the ShippingMethod's configuration it should have the ``first_unit_cost``, ``additional_unit_cost`` and ``additional_unit_limit`` configured - it is useful if you want to have different cost for teh fist item than for the rest.
+* **PerUnitRateCalculator** - returns the ``amount`` from the ShippingMethod's configuration multiplied by the ``units`` count.
+* **VolumeRateCalculator** - returns the ``amount`` from the ShippingMethod's configuration multiplied by the ``units`` count divided by the configured ``division``.
+* **WeightRateCalculator** - returns the ``fixed`` from the ShippingMethod's configuration multiplied plus the ``variable`` multiplied by the ``weight`` of the shipment and divided by the ``division`` from the configuration.
+
+
+Shipment Calculator Configuration
+---------------------------------
+
+As it was mentioned above the calculators may have different configuration in an array. Such keys are available:
```
Comment (pjedrzejewski) [inline docs/book/shipments.rst]: Different keys are available for different calculators, we should not list them here. We will have reference with all shipping calculators and their configurations later. Please remove this part.

File: docs/book/shipments.rst [hunk 7]
```diff
@@ -4,49 +4,139 @@
...
+        'additional_unit_cost' => 2,
+        'additional_unit_limit` => 10
+    ];
+
+ProductVariant Configuration
+----------------------------
 
-...
+In order to be able to calculate shipping costs basing on the volume and weight of products in an order
+the ProductVariant has the ``depth``, ``width``, ``height`` and ``weight`` fields.
```
Comment (pjedrzejewski) [inline docs/book/shipments.rst]: 👍

File: docs/book/shipments.rst (hunk 4 above)
Comment (TheMadeleine) [inline docs/book/shipments.rst]: Added.

**Comment by pjedrzejewski** [comment]
Thank you Magda!

**Pull request merged**